### ISSUE TITLE
Error in solution to Exercise 2.2

### ISSUE THREAD

**Issue by earthwuyang** [vojtamolda/reinforcement-learning-an-introduction#9]
I think the calculation of the rewards of each arm in each step is not sample-average according to the formula given in the book.

**Comment by earthwuyang** [comment]
![微信图片_20220213151810](https://user-images.githubusercontent.com/53375253/153743193-05a2f55e-def3-4002-8ea4-f5a9a12a5d29.jpg)
Thank you! I might also be wrong. I would be thankful if you could point out my error if it's my mistake.

**Comment by vojtamolda** [comment]
I recalculated my solution from scratch and you're right! I'm not sure how I arrived at my solution...

Anyway, to make the calculation clear, I think, one has to keep vector with sum of rewards for each action `Si` together with a counter `ni`. The counter tracks the number of times a particular state has been taken. The action-value is then calculated as a ratio of sum of rewards over the number of visits `Qi = Si/ni`.

Here's the full updated solution (it exactly matches yours):

<img width="426" alt="Screen Shot 2022-02-14 at 16 26 38" src="https://user-images.githubusercontent.com/8467740/153956982-4d7ca8a5-f8d3-4442-8422-a9131e37e514.png">
.
